src/main.rs

```diff
@@ -38,9 +40,16 @@ impl Callback for Client {
     }
 }
 
+#[derive(Deserialize)]
+struct MainConfig {
+    identity_store: String
+}
```

Comment:
String is the wrong type. It should be a custom enum.

```suggestion
    identity_store: IdentityStoreType
}

#[derive(Deserialize)]
enum IdentityStoreType {
  Ldap,
  Json,
}
```